pravega/pravega-operator
# Issue 130: Ability to specify container resources

### Change log description

This PR will bring the following changes:
- Set adequate default resource requests and limits for Pravega components
- Expose those resource setting to allow custom configurations
- Replace static `Xmx` JVM option with the dynamic, Docker-aware flags `-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap` which will set the heap memory size to the given Pod memory limit. (Ref: https://blogs.oracle.com/java-platform-group/java-se-support-for-docker-cpu-and-memory-limits)

At this moment, the PR is being used for testing with the goal of finding the suitable values for resource requests and limits.

### Purpose of the change
Fix #130 

### How to test the change
When defining the Pravega cluster specification in the YAML file, include the following options to set custom resource requests and limits for each Pravega component.

```
apiVersion: "pravega.pravega.io/v1alpha1"
kind: "PravegaCluster"
metadata:
  name: "example"
spec:
  ...
  bookkeeper:
    ...
    resources:
      requests:
        memory: "3Gi"
        cpu: "1000m"
      limits:
        memory: "5Gi"
        cpu: "2000m"

  pravega:
    ...
    controllerResources:
      requests:
        memory: "1Gi"
        cpu: "1000m"
      limits:
        memory: "3Gi"
        cpu: "2000m"

    segmentStoreResources:
      requests:
        memory: "3Gi"
        cpu: "1000m"
      limits:
        memory: "5Gi"
        cpu: "2000m"
```

You can verify that those values have been correctly set by looking at the Pods description.

---
Signed-off-by: Adrián Moreno <adrian@morenomartinez.com>
